kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/base/OperationSupport.java

```diff
@@ -592,6 +592,9 @@ protected Response retryWithExponentialBackoff(OkHttpClient client, Request requ
         if (numRetries < requestRetryBackoffLimit && response.code() >= 500) {
           retryInterval = retryIntervalCalculator.getInterval(numRetries);
           LOG.debug("HTTP operation on url: {} should be retried as the response code was {}, retrying after {} millis", request.url(), response.code(), retryInterval);
+          if (response.body() != null) {
+            response.body().close();
+          }
```

Comment:
The response implements Closable, the OkHttp leak message refers to that closable method not getting invoked.

https://github.com/fabric8io/kubernetes-client/issues/3561#issuecomment-959685922